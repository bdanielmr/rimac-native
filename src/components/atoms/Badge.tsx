import { LinearGradient } from "expo-linear-gradient";
import React from "react";
import styled from "styled-components/native";

const BadgeContainer = styled.View`
  align-self: flex-start;
  border-radius: 4px;
  overflow: hidden;
  margin-bottom: 8px;
`;

const BadgeGradient = styled(LinearGradient)`
  padding: 6px 10px;
  border-radius: 4px;
`;

const BadgeText = styled.Text`
  color: #03050F;
  font-weight: 700;
  font-size: 12px;
`;

interface BadgeProps {
  children: React.ReactNode;
  testID?: string;
}

export const Badge: React.FC<BadgeProps> = ({ children, testID }) => (
  <BadgeContainer testID={testID}>
    <BadgeGradient
      colors={["#00F4E2", "#00FF7F"]}
      start={{ x: 0, y: 0 }}
      end={{ x: 1, y: 0 }}
    >
      <BadgeText>{children}</BadgeText>
    </BadgeGradient>
  </BadgeContainer>
);