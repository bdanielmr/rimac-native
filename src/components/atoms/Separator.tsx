import styled from "styled-components/native";

export const Separator = styled.View`
  height: 1px;
  background-color: #e5e7eb;
  width: 100%;
  margin-bottom: 24px;
`;