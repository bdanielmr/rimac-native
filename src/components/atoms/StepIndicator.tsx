import React from "react";
import styled from "styled-components/native";

const StepContainer = styled.View`
  flex-direction: row;
  align-items: center;
  justify-content: center;
  padding: 20px 16px;
`;

const StepCircle = styled.View<{ active: boolean; completed: boolean }>`
  width: 24px;
  height: 24px;
  border-radius: 12px;
  background-color: ${(props: any) =>
    props.active ? "#4F4FFF" : props.completed ? "transparent" : "transparent"};
  align-items: center;
  justify-content: center;
  border-width: 1px;
  border-color: ${(props: any) =>
    props.active ? "#4F4FFF" : props.completed ? "#7981B2" : "#7981B2"};
`;

const StepNumber = styled.Text<{ active: boolean; completed: boolean }>`
  color: ${(props: any) => 
    props.active ? "white" : props.completed ? "#7981B2" : "#7981B2"};
  font-weight: 700;
  font-size: 12px;
`;

const StepLabel = styled.Text<{ active: boolean; completed: boolean }>`
  font-size: 16px;
  font-weight: ${(props: any) => (props.active ? "700" : "400")};
  color: ${(props: any) => 
    props.active ? "#141938" : props.completed ? "#7981B2" : "#7981B2"};
`;

const SeparatorIndicator = styled.Text<{ completed: boolean }>`
  color: ${(props: any) => (props.completed ? "#7981B2" : "#4F4FFF")};
  margin: 0 16px;
  font-size: 20px;
`;

const StepWrapper = styled.View`
  flex-direction: row;
  align-items: center;
  gap: 12px;
`;

interface Step {
  number: number;
  label: string;
}

interface StepIndicatorProps {
  steps: Step[];
  currentStep: number;
}

export const StepIndicator: React.FC<StepIndicatorProps> = ({ steps, currentStep }) => (
  <StepContainer>
    {steps.map((step, index) => {
      const isActive = currentStep === step.number;
      const isCompleted = currentStep > step.number;
      
      return (
        <React.Fragment key={step.number}>
          <StepWrapper>
            <StepCircle active={isActive} completed={isCompleted}>
              <StepNumber active={isActive} completed={isCompleted}>
                {step.number}
              </StepNumber>
            </StepCircle>
            <StepLabel active={isActive} completed={isCompleted}>
              {step.label}
            </StepLabel>
          </StepWrapper>
          {index < steps.length - 1 && (
            <SeparatorIndicator completed={isCompleted}>
              - - - -
            </SeparatorIndicator>
          )}
        </React.Fragment>
      );
    })}
  </StepContainer>
);