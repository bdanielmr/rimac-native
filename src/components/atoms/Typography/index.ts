import styled from "styled-components/native";

export const Title = styled.Text`
  font-size: 32px;
  font-weight: 700;
  color: #03050f;
  margin-bottom: 8px;
  line-height: 40px;
  max-width: 352px;
`;

export const MobileTitle = styled.Text`
  font-size: 28px;
  font-weight: 700;
  color: #03050f;
  margin-bottom: 8px;
  line-height: 36px;
`;

export const Subtitle = styled.Text`
  color: #03050f;
  font-size: 14px;
  line-height: 20px;
  margin-bottom: 15px;
  max-width: 352px;
  font-weight: 600;
`;