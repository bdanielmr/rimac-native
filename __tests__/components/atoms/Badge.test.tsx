import { render } from '@testing-library/react-native';
import React from 'react';
import { Badge } from '../../../src/components/atoms/Badge';

describe('Badge Component', () => {
  it('renders correctly with text content', () => {
    const { getByText } = render(<Badge>Promoción</Badge>);
    expect(getByText('Promoción')).toBeTruthy();
  });

  it('renders with different text content', () => {
    const { getByText } = render(<Badge>Nuevo</Badge>);
    expect(getByText('Nuevo')).toBeTruthy();
  });

  it('renders multiple badges independently', () => {
    const { getByText } = render(
      <>
        <Badge>Promoción</Badge>
        <Badge>Destacado</Badge>
        <Badge>Oferta</Badge>
      </>
    );
    
    expect(getByText('Promoción')).toBeTruthy();
    expect(getByText('Destacado')).toBeTruthy();
    expect(getByText('Oferta')).toBeTruthy();
  });

  it('renders with numeric content', () => {
    const { getByText } = render(<Badge>100</Badge>);
    expect(getByText('100')).toBeTruthy();
  });

  it('renders with long text content', () => {
    const longText = 'Este es un texto muy largo para el badge';
    const { getByText } = render(<Badge>{longText}</Badge>);
    expect(getByText(longText)).toBeTruthy();
  });

  it('can be found by testID when provided', () => {
    const { getByTestId } = render(
      <Badge testID="promo-badge">Promoción</Badge>
    );
    expect(getByTestId('promo-badge')).toBeTruthy();
  });

  it('renders with special characters', () => {
    const { getByText } = render(<Badge>50% OFF 🎉</Badge>);
    expect(getByText('50% OFF 🎉')).toBeTruthy();
  });

  it('matches snapshot', () => {
    const { toJSON } = render(<Badge>Promoción</Badge>);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders as Text component', () => {
    const { getByText } = render(<Badge>Test</Badge>);
    const badge = getByText('Test');
    expect(badge.type).toBe('Text');
  });
});