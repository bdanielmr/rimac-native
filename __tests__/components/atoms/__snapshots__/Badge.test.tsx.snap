// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Component matches snapshot 1`] = `
<View
  style={
    {
      "alignSelf": "flex-start",
      "borderBottomLeftRadius": 4,
      "borderBottomRightRadius": 4,
      "borderTopLeftRadius": 4,
      "borderTopRightRadius": 4,
      "marginBottom": 8,
      "overflow": "hidden",
    }
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    colors={
      [
        4278252770,
        4278255487,
      ]
    }
    endPoint={
      [
        1,
        0,
      ]
    }
    startPoint={
      [
        0,
        0,
      ]
    }
    style={
      {
        "borderBottomLeftRadius": 4,
        "borderBottomRightRadius": 4,
        "borderTopLeftRadius": 4,
        "borderTopRightRadius": 4,
        "paddingBottom": 6,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 6,
      }
    }
  >
    <Text
      style={
        {
          "color": "#03050F",
          "fontSize": 12,
          "fontWeight": "700",
        }
      }
    >
      Promoci√≥n
    </Text>
  </ViewManagerAdapter_ExpoLinearGradient>
</View>
`;
